core/controllers/base.py

```diff
@@ -295,8 +299,71 @@ def dispatch(self):
                 self.handle_exception(e, self.app.debug)
                 return
 
+        try:
+            self.validate_args_schema()
+        except Exception as e:
+            self.handle_exception(e, self.app.debug)
+            return
+
         super(BaseHandler, self).dispatch()
 
+    def validate_args_schema(self):
+        """Validates schema for controller layer handler class arguments.
+
+        Raises:
+            InvalidInputException. Schema validation failed.
+            NotImplementedError. Schema is not provided in handler class.
+        """
+        handler_class_name = self.__class__.__name__
+        request_method = self.request.environ['REQUEST_METHOD']
+        handler_args = self.payload
+        url_path_args = self.request.route_kwargs
+        non_schema_handlers = payload_validator.NON_SCHEMA_HANDLERS
```

Comment:
NON_SCHEMA_HANDLERS is a list of handlers for which schema validation will not be performed.

Reply:
Seems like HANDLER_CLASS_NAMES_WITH_NO_SCHEMA to me.

(A handler and its class name are different things; please name things specifically. This applies to other names you have introduced as well.)